src/ucp/api/ucp.h

```diff
@@ -331,6 +331,47 @@ enum ucp_ep_close_mode {
 };
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief UCP performance fields and flags
+ *
+ * The enumeration allows specifying which fields in @ref ucp_ep_perf_attr_t are
+ * present and operation flags are used. It is used to enable backward
+ * compatibility support.
+ */
+enum ucp_ep_perf_attr_field {
+    /** Enables @ref ucp_ep_perf_attr_t::message_size */
+    UCP_EP_ATTR_FIELD_MESSAGE_SIZE       = UCS_BIT(0),
+
+    /** Enables @ref ucp_ep_perf_attr_t::local_memory_type */
+    UCP_EP_ATTR_FIELD_LOCAL_MEMORY_TYPE  = UCS_BIT(1),
+
+    /** Enables @ref ucp_ep_perf_attr_t::remote_memory_type */
+    UCP_EP_ATTR_FIELD_REMOTE_MEMORY_TYPE = UCS_BIT(2),
+
+    /** Enables @ref ucp_ep_perf_attr_t::operation */
+    UCP_EP_ATTR_FIELD_OPERATION          = UCS_BIT(3),
+
+    /** Enables @ref ucp_ep_perf_attr_t::bandwidth */
+    UCP_EP_ATTR_FIELD_BANDWIDTH          = UCS_BIT(4)
+};
+
+
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief All existing UCP operations
+ * 
+ * This enumeration defines all available UCP operations.
+ */
+typedef enum ucp_ep_op {
```

Comment:
Maybe using `ucp_feature` would be appropriate here, instead of creating a new enum? Unsure about that..

Reply:
let's use ucp_ep_operation_t

Reply:
Are you sure?
My logic was to be consistent with types like `ucp_op_attr_t`, `ucp_atomic_fetch_op_t`, etc. -  which use `_op_` and not `_operation_`.

Reply:
ok 👍

Reply:
@gleon99 maybe rename to ucp_ep_op_type_t
UCP_OP_TYPE_TAG
and
`ucp_ep_op_type_t op_type` in the struct?

because TAG is {tag_send,tag_send_sync}; RMA is {put,get} - each value is >1 specific op